services/src/test/scala/cromwell/services/instrumentation/AsynchronousThrottlingGaugeMetricActorSpec.scala

```diff
@@ -0,0 +1,68 @@
+package cromwell.services.instrumentation
+
+import akka.testkit.{ImplicitSender, TestFSMRef, TestProbe}
+import cats.data.NonEmptyList
+import cromwell.core.TestKitSuite
+import cromwell.core.instrumentation.InstrumentationPrefixes
+import cromwell.services.instrumentation.AsynchronousThrottlingGaugeMetricActor.{CalculateMetricValue, MetricCalculationInProgress, MetricValue, WaitingForMetricCalculationRequestOrMetricValue}
+import cromwell.services.instrumentation.InstrumentationService.InstrumentationServiceMessage
+import org.scalatest.{FlatSpecLike, Matchers}
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.Promise
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+class AsynchronousThrottlingGaugeMetricActorSpec extends TestKitSuite with FlatSpecLike with ImplicitSender with Matchers with Eventually {
+
+  private val defaultTimeout = 5 seconds
+
+  it should "follow the proper state transition path when requested to calculate metric value and calculation succeeded" in {
```

Comment:
I think there's two more possible state transitions flow worth testing:

First, a metric value arrives during the calculation:
* The actor receives a calculate metric request
* The actor receives a `MetricValue`
* The original calculation returns its value

Second, a second calculation request arrives while the first calculation is ongoing:
* The actor receives a calculate metric request
* The actor receives a `MetricValue`
* The original calculation returns its value